client/app/assets/less/redash/ant.less

```diff
@@ -36,3 +42,35 @@
 .ant-dropdown-in-bootstrap-modal {
   z-index: 1050;
 }
+
+// Make Ant labels more Redash like
+.ant-form-item-label label {
+  color: inherit;
+  white-space: initial;
+  line-height: initial;
+  text-align: left;
+}
+
+// Fix ant input number showing duplicate arrows
+.ant-input-number-input::-webkit-outer-spin-button,
+.ant-input-number-input::-webkit-inner-spin-button {
+  -webkit-appearance: none;
+  margin: 0;
+}
+
+.ant-input-number-input {
+  -moz-appearance: textfield;
+}
+
+// Fix for text becoming black on disabled buttons
+.btn-danger, .btn-primary, .btn-success, .btn-info {
+  &[disabled] {
+    color: @white !important;
+  }
+}
```

Comment:
Can this be avoided by converting the related buttons from bootstrap to Ant?

Reply:
I think so, they are actually using Ant, however the color is still given through a `class` (bootstrap format) argument from the object. Basically what's needed is to rethink this (Data-sources angular page): 
```javascript
$scope.actions = [
    { name: 'Delete', class: 'btn-danger', callback: deleteDataSource },
    {
      name: 'Test Connection', class: 'btn-default pull-right', callback: testConnection, disableWhenDirty: true,
    },
  ]
```
Then we would either need to map this in DynamicForm Transition or actually change it.

My first idea on that was to simply use css to fix it till we actually migrate the parent. But thinking better, this would not be the best since I would be generating some debt :thinking:.

Reply:
I agree, that's certainly sth I would change when handling the Pages.

Reply:
I think this should do the trick:
```js
// show.js
$scope.actions = [
    { name: 'Delete', type: 'danger', callback: deleteDataSource },
    …
]
```
```jsx
// DynamicForm.jsx
renderActions() {
  return this.props.actions.map((action) => {
    …
    const actionProps = {
      key: action.name,
      htmlType: 'button',
      className: `${action.class} m-t-10`,
      type: action.type, // <--- defaults to 'default'
      …
    };

    return (<Button …>…</Button>);
  });
}
```

Reply:
Or perhaps just give it the class `ant-btn-danger` instead of `btn-warning`?

Reply:
Well, remove the `class` prop was already in my plans, so I used a `type` and a `purgeRight` prop. However, my guess was wrong and the color for disabled danger buttons keeps turning to black...